src/core/evaluator.js

```diff
@@ -1406,11 +1407,20 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
         textContentItem.str.length = 0;
       }
 
+      function enqueueChunk() {
+        if (textContent.items.length > 0) {
```

Comment:
Nit: To avoid recomputing the length, I'd suggest adding e.g. `let length = textContent.items.length;` and using that here and below.